presc/copies/continuous.py

```diff
@@ -0,0 +1,185 @@
+            samples used so far for the incremental training of the copy, and
+            evaluates the performance of the resulting copy.
+        test_data : presc.dataset.Dataset
+            Dataset with the test data that will be used to evaluate the copy
+            after each training iteration.
+    """
+
+    def __init__(
+        self, classifier_copy, data_stream, fit_kwargs={}, verbose=False, test_data=None
+    ):
+        Thread.__init__(self)
+        self.classifier_copy = classifier_copy
+        self.data_stream = data_stream
+        self.fit_kwargs = fit_kwargs
+        self.verbose = verbose
+        self.test_data = test_data
+
+        self.iterations = 0
+        self.n_samples = 0
+
+        if isinstance(self.classifier_copy.copy, Pipeline):
+            self.is_pipeline = True
+        else:
+            self.is_pipeline = False
+
+    def run(self):
+        """Function to start the copying of the continuous online classifier."""
+        self.continuous_copy_run = True
+
+        self.partial_fit_ok = check_partial_fit(self.classifier_copy.copy)
+        if not self.partial_fit_ok:
+            print("Sorry, not all those transformers/estimators have 'partial_fit'.")
+
+        while self.continuous_copy_run and self.partial_fit_ok:
+            # Get data
+            data_block = self.data_stream.get()
+            n_samples_block = len(data_block.df)
+
+            # If pipeline, train elements of pipeline sequentially,
+            # and use transformed data to fit the next transformer or estimator
+            if self.is_pipeline:
+                n_estimators = len(self.classifier_copy.copy.named_steps)
+                X = data_block.features.copy()
+                for index, element in enumerate(self.classifier_copy.copy.named_steps):
+                    self.classifier_copy.copy[element].partial_fit(
+                        X, y=data_block.labels, **self.fit_kwargs[element]
+                    )
+                    # Transform data for next estimator except for the last one
+                    if index + 1 < n_estimators:
+                        X = self.classifier_copy.copy.named_steps[element].transform(X)
+
+            # If single classifier, train model with this block
+            else:
+                self.classifier_copy.copy.partial_fit(
+                    data_block.features, data_block.labels, **self.fit_kwargs
+                )
+            self.iterations += 1
+            self.n_samples += n_samples_block
+
+            if self.verbose:
+                print("\nIteration: ", self.iterations)
+                print("Samples: ", self.n_samples, "\n")
+
+                self.classifier_copy.evaluation_summary(
+                    test_data=self.test_data, synthetic_data=data_block
+                )
+
+    def stop(self):
+        """Function to stop the continuous copying of the online classifier."""
+        self.continuous_copy_run = False
```

Comment:
Similarly as for SyntheticDataStreamer, the thread blocks on `self.data_stream.get()` if the queue is empty, and won't actually stop until it is able to get an item from the queue. Maybe push a sentinel item to the queue here (`None` or some other special value), and then check for it in the while loop, skipping the rest of the loop if `data_block` is that value.

Reply:
Thanks! Fixed.